Minor commit validation fixes

ironcladlou opened the pull request:
Minor commit validation fixes.

* Update hack/cherry-pick.sh to produce valid commit summaries
* Improve validation examples
* Remove validation debugging output
* Ignore bump commits (for now)
* Allow unqualified non-kube repo references

ironcladlou commented:
@deads2k

deads2k commented:
@ironcladlou do you think there's a reasonable case for making `cherry-pick.sh` into a small go command?

ironcladlou commented:
>do you think there's a reasonable case for making cherry-pick.sh into a small go command?

I don't have enough experience with it to say at this moment...

deads2k commented on cmd/rebasehelpers/commitchecker/validate.go:
```diff
@@ -28,11 +28,11 @@ UPSTREAM commits are validated against the following regular expression:
-  UPSTREAM: 12345: An origin change
-  UPSTREAM: docker/docker: 12345: A docker change
+  UPSTREAM: 12345: A kube fix
+  UPSTREAM: github.com/coreos/etcd: 12345: An etcd fix
```
Can we update the detection code to make it just `coreos/etcd`?  I know there's a risk of conflicts, but its a really small risk.

ironcladlou replied:
I'm not yet seeing how we can without maintaining list of every base account name we depend on (e.g. coreos).

liggitt replied:
make sure there are at least two segments, and that all the changed files under godeps match "Godeps/_workspace/src/" ... optional single segment (lets us skip gopkg.in or github.com, etc) ... "segment1/segment2/" ...?

deads2k replied:
> I'm not yet seeing how we can without maintaining list of every base account name we depend on (e.g. coreos).

You've already got a list of how to handle things like github.com, k8s.io, etc, so I think you can go from godep file change to expected repo name with no difficulty.  The check for ensuring that they're updating the correct godep would be relaxed to allow "*/cores/etcd/*".  There's a risk of false negatives, but its pretty darn low.

ironcladlou replied:
Yeah, I think that will work. Let me tinker with that.

deads2k commented on hack/cherry-pick.sh:
```diff
@@ -80,9 +81,14 @@ if [[ $? -ne 0 ]]; then
 
+commit_message="UPSTREAM: $pr: Cherry-picked"
+if [ "$repo" != "k8s.io/kubernetes" ]; then
+  commit_message="UPSTREAM: $repo: $pr: Cherry-picked"
```
@liggitt here's your chance.  Shall we update to indicate which ones were merged at the time of picking?

deads2k commented:
`UPSTREAM: Removed obsoleted godep code.google.com/p/go-uuid/uuid` should be allowed, but isn't

deads2k commented:
> It should only fail on the name

On second thought, maybe we should have an "unbump"?  @miminar may need to rename is unbump to a "bump" for now

deads2k commented on cmd/rebasehelpers/commitchecker/commitchecker.go:
```diff
@@ -21,9 +21,19 @@ func main() {
+	nonbumpCommits := []util.Commit{}
+	for _, commit := range commits {
+		if commit.DeclaresUpstreamChange() &&
+			!strings.Contains(strings.ToLower(commit.Summary), "bump") {
```
`!strings.HasPrefix(commit, "bump")`

ironcladlou replied:
Won't the prefix be "UPSTREAM:"?

deads2k replied:
> Won't the prefix be "UPSTREAM:"?

bumps start with `bump`

ironcladlou replied:
bumpkin

ironcladlou replied:
Prefix "UPSTREAM: bump"?

ironcladlou replied:
Now checking for prefix "bump("

ironcladlou commented:
>

UPSTREAM: Removed obsoleted godep code.google.com/p/go-uuid/uuid should be allowed, but isn't. It's failing for "problem with file "Godeps/Godeps.json" in commit d233e77: file doesn't appear to be a Godeps workspace path" It should only fail on the name

Updated to make summary name validation the first in the list, which makes sense for other reasons (if the summary's invalid we can't get a declared repo out of it, etc.)

deads2k commented:
One comment on prefix, then we better get this in to avoid bump issues.

deads2k commented:
squash and merge at will

ironcladlou commented:
[merge]

ironcladlou commented:
Integration test flake, [merge] again

miminar commented:
Thanks guys! :+1: for `unbump`.
